device/common/include/traccc/edm/device/sort_key.hpp

```diff
@@ -0,0 +1,34 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/edm/track_parameters.hpp"
+
+namespace traccc::device {
+
+struct sort_key {
+    traccc::scalar key;
+};
+
+TRACCC_HOST_DEVICE
+inline sort_key get_sort_key(const bound_track_parameters& params) {
+    // key = |theta - pi/2|
+    return sort_key{math::abs(params.theta() - constant<traccc::scalar>::pi_2)};
+}
```

Comment:
Though if we keep `sort_key` as-is, couldn't this become a constructor on that class? :thinking: That would give the code a more C\+\+-ie interface.